tests/suite/known_triples.rs

```diff
@@ -0,0 +1,92 @@
+use std::collections::BTreeSet;
+
+use platforms::Platform;
+
+#[test]
+fn gen_known_triples() {
+    let output = "src/dist/triple/known.rs";
+    let existing = std::fs::read_to_string(output).unwrap();
+    assert_eq!(known_triples(), existing);
```

Comment:
The developer ergonomics are nicer if you can move the write the new output into the file in the failure case here, so that a human user can easily commit the updates and move on. See the examples I linked before.

Reply:
@djc I have adjusted the test accordingly, and I have tested the in-place update to confirm that it's actually working.